test_runner/random_test_runner/Readme.md

```diff
@@ -0,0 +1,213 @@
+
+## How to run
+
+First complete build described in [How to build](#how-to-build) section for autoware architecture you are using.
+
+Then, in first terminal run: 
+
+```shell
+rviz2 -d $(ros2 pkg prefix random_test_runner)/share/random_test_runner/rviz/random_test.rviz
+```
+
+In second:
+
+```shell
+ros2 launch random_test_runner random_test.launch.py
+```
+
+Which will run random tests with default paramters. You should see several npcs spawned in random locations around the ego vehicle, which will move on random path.
+
+After test is completed see `/tmp` directory. Among others, there will be two files:
+1. `result.junit.xml` - test result file with information about encountered errors
+2. `result.yaml` - yaml file that can be used to replay tests
+
+Exemplary content of those files can be seen below:
+
+`result.junit.xml`:
+```xml
+<?xml version="1.0"?>
+<testsuites failures="0" errors="6" tests="6">
+  <testsuite name="random_test14699091947383883875" failures="0" errors="6" tests="6">
+    <testcase name="4">
+      <error type="stand still" message="Ego seems to be stuck" />
+    </testcase>
+    <testcase name="2">
+      <error type="collision" message="npc2 and ego collided at 6.249999999999986s" />
+      <error type="stand still" message="Ego seems to be stuck" />
+    </testcase>
+    <testcase name="3">
+      <error type="collision" message="npc1 and ego collided at 17.25000000000011s" />
+      <error type="timeout" message="Ego failed to reach goal within timeout" />
+    </testcase>
+    <testcase name="1">
+      <error type="stand still" message="Ego seems to be stuck" />
+    </testcase>
+    <testcase name="0" />
+  </testsuite>
+</testsuites>
+```
+
+`result.yaml`:
+```yaml
+random_test14699091947383883875:
+  name: video_test
+  map_name: kashiwanoha_map
+  ego_goal_s: 5.000000000000000000
+  ego_goal_lanelet_id: -1
+  ego_goal_partial_randomization: false
+  ego_goal_partial_randomization_distance: 30
+  npc_count: 10
+  npc_min_speed: 0.500000000000000000
+  npc_max_speed: 3.000000000000000000
+  npc_min_spawn_distance_from_ego: 10.000000000000000000
+  npc_max_spawn_distance_from_ego: 50.000000000000000000
+  test_cases:
+    - seed: 1281242544
+    - seed: 3644198185
+    - seed: 2673087374
+    - seed: 1312244741
+    - seed: 2518911638
+```
+
+## How to replay
+
+Prerequisites:
+1. Build as instructed in [How to build](#how-to-build)
+2. Acquire `result.yaml` file:
+   1. Either by running test as stated in [How to run](#how-to-run) part of instruction
+   2. Receiving it from someone who already ran it
+3. Place `result.yaml` in `<some_directory>` IMPORTANT: Do not change filename
+4. Execute:
+ 
+```shell
+ros2 launch random_test_runner random_test.launch.py input_dir:=<some_directory>
+```
+
+Random test runner will load `result.yaml` file and rerun test.
+
+## Running with unity
+
+To run with unity: 
+1. Run unity project that supports `scenario_simulator` integration
```

Comment:
Do you have some examples of a "project that supports `scenario_simulator` integration"? Or some instruction on how to integrate unity with ss2? If I would like to run it with unity I have no clue what that kind of project needs to have